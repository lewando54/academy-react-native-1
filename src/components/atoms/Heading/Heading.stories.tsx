/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react'

import Heading from './Heading'
import { create } from 'react-test-renderer'

const meta = {
    component: Heading,
    argTypes: {
        level: { control: 'select', options: [1, 2, 3, 4, 5, 6] },
        children: { control: 'text' }
    }
} satisfies Meta<typeof Heading>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
    args: {
        children: 'This is a heading',
        level: 1,
        testId: 'test'
    }
}