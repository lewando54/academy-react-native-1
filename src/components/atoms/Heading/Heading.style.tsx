import { StyleSheet } from 'react-native'

export default StyleSheet.create({
    base:{
        fontWeight: '700',
        textTransform: 'uppercase',
        textAlign: 'left',
        margin: 0,
        width: '100%'
    }
})