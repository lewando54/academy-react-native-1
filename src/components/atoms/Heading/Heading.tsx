import { Text } from 'react-native'
import headingStyle from './Heading.style'

export interface IHeadingProps {
  level?: 1 | 2 | 3 | 4 | 5 | 6
  children?: React.ReactNode | string
  testId: string
}

function Heading ({ level = 1, children, testId }: IHeadingProps): React.ReactElement {
  return (
    <Text testID={testId} style={[
        headingStyle.base,
        {fontSize: 16 / (level / 2) + 6}
        ]}>
      {children}
    </Text>
  )
}

export default Heading