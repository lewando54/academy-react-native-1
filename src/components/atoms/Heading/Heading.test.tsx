import React from 'react'
import * as renderer from 'react-test-renderer'
import Heading from './Heading'

describe('Heading', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Heading testId='heading'>Testing</Heading>)
    expect(tree).toMatchSnapshot()
  })

  it('should render innerHTML', () => {
    const tree = renderer.create(<Heading testId='heading'>Testing</Heading>)
    expect(tree).toMatchSnapshot()
  })

  it('should render <h1> tag by default', () => {
    const tree = renderer.create(<Heading testId='heading'>Testing</Heading>)
    expect(tree).toMatchSnapshot()
  })

  it('should render <h1> tag', () => {
    const tree = renderer.create(<Heading level={1} testId='heading'>Testing</Heading>)
    expect(tree).toMatchSnapshot()
  })

  it('should render <h2> tag', () => {
    const tree = renderer.create(<Heading level={2} testId='heading'>Testing</Heading>)
    expect(tree).toMatchSnapshot()
  })

  it('should render <h3> tag', () => {
    const tree = renderer.create(<Heading level={3} testId='heading'>Testing</Heading>)
    expect(tree).toMatchSnapshot()
  })

  it('should render <h4> tag', () => {
    const tree = renderer.create(<Heading level={4} testId='heading'>Testing</Heading>)
    expect(tree).toMatchSnapshot()
  })

  it('should render <h5> tag', () => {
    const tree = renderer.create(<Heading level={5} testId='heading'>Testing</Heading>)
    expect(tree).toMatchSnapshot()
  })

  it('should render <h6> tag', () => {
    const tree = renderer.create(<Heading level={6} testId='heading'>Testing</Heading>)
    expect(tree).toMatchSnapshot()
  })
})
