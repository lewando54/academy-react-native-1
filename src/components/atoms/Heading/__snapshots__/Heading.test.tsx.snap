// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render <h1> tag 1`] = `
<Text
  style={
    [
      {
        "fontWeight": "700",
        "margin": 0,
        "textAlign": "left",
        "textTransform": "uppercase",
        "width": "100%",
      },
      {
        "fontSize": 38,
      },
    ]
  }
  testID="heading"
>
  Testing
</Text>
`;

exports[`Heading should render <h1> tag by default 1`] = `
<Text
  style={
    [
      {
        "fontWeight": "700",
        "margin": 0,
        "textAlign": "left",
        "textTransform": "uppercase",
        "width": "100%",
      },
      {
        "fontSize": 38,
      },
    ]
  }
  testID="heading"
>
  Testing
</Text>
`;

exports[`Heading should render <h2> tag 1`] = `
<Text
  style={
    [
      {
        "fontWeight": "700",
        "margin": 0,
        "textAlign": "left",
        "textTransform": "uppercase",
        "width": "100%",
      },
      {
        "fontSize": 22,
      },
    ]
  }
  testID="heading"
>
  Testing
</Text>
`;

exports[`Heading should render <h3> tag 1`] = `
<Text
  style={
    [
      {
        "fontWeight": "700",
        "margin": 0,
        "textAlign": "left",
        "textTransform": "uppercase",
        "width": "100%",
      },
      {
        "fontSize": 16.666666666666664,
      },
    ]
  }
  testID="heading"
>
  Testing
</Text>
`;

exports[`Heading should render <h4> tag 1`] = `
<Text
  style={
    [
      {
        "fontWeight": "700",
        "margin": 0,
        "textAlign": "left",
        "textTransform": "uppercase",
        "width": "100%",
      },
      {
        "fontSize": 14,
      },
    ]
  }
  testID="heading"
>
  Testing
</Text>
`;

exports[`Heading should render <h5> tag 1`] = `
<Text
  style={
    [
      {
        "fontWeight": "700",
        "margin": 0,
        "textAlign": "left",
        "textTransform": "uppercase",
        "width": "100%",
      },
      {
        "fontSize": 12.4,
      },
    ]
  }
  testID="heading"
>
  Testing
</Text>
`;

exports[`Heading should render <h6> tag 1`] = `
<Text
  style={
    [
      {
        "fontWeight": "700",
        "margin": 0,
        "textAlign": "left",
        "textTransform": "uppercase",
        "width": "100%",
      },
      {
        "fontSize": 11.333333333333332,
      },
    ]
  }
  testID="heading"
>
  Testing
</Text>
`;

exports[`Heading should render correctly 1`] = `
<Text
  style={
    [
      {
        "fontWeight": "700",
        "margin": 0,
        "textAlign": "left",
        "textTransform": "uppercase",
        "width": "100%",
      },
      {
        "fontSize": 38,
      },
    ]
  }
  testID="heading"
>
  Testing
</Text>
`;

exports[`Heading should render innerHTML 1`] = `
<Text
  style={
    [
      {
        "fontWeight": "700",
        "margin": 0,
        "textAlign": "left",
        "textTransform": "uppercase",
        "width": "100%",
      },
      {
        "fontSize": 38,
      },
    ]
  }
  testID="heading"
>
  Testing
</Text>
`;
